regex_test/regexpatterns.go

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright (c) 2020-present Snowplow Analytics Ltd.
+ * All rights reserved.
+ *
+ * This software is made available by Snowplow Analytics, Ltd.,
+ * under the terms of the Snowplow Limited Use License Agreement, Version 1.0
+ * located at https://docs.snowplow.io/limited-use-license-1.0
+ * BY INSTALLING, DOWNLOADING, ACCESSING, USING OR DISTRIBUTING ANY PORTION
+ * OF THE SOFTWARE, YOU AGREE TO THE TERMS OF SUCH LICENSE AGREEMENT.
+ */
+
+package regexpatterns
+
+import (
+	"regexp"
+)
+
+
+func reMatch(re *regexp.Regexp, line string, numGoroutines int) {
+	c := make(chan int)
+	for i := 0; i < numGoroutines; i++ {
+		go func() {
+			re.MatchString(line)
+			c <- 1
```

Comment:
Mentioned when we discussed yesterday - channels are fairly expensive. What I didn't think to mention is you can use a [waitgroup](https://stackoverflow.com/a/18207832) to do the same for cheaper - which would reduce the amount of possible skewing factors in the benchmark.

Not asking you to change it here - just following up with relevant info that I neglected to mention yesterday. I think what you have here measures things equally enough for our purposes.